inline-r/src/Language/R/HExp.hsc

```diff
@@ -307,56 +307,60 @@ peekHExp s = do
         -- fields actually always contain fields of form ANYSXP. This has no
         -- operational significance - it's only a way to bypass what's
         -- impossible to prove.
-        coerceAny :: SEXP s a -> SEXP s 'R.Any -- '
+        coerceAny :: SEXP s b -> SEXP s 'R.Any -- '
         coerceAny = R.unsafeCoerce
 
-        sptr = R.unsexp s
+        coerceAnySome :: SomeSEXP s -> SEXP s 'R.Any -- '
+        coerceAnySome (SomeSEXP s1) = coerceAny s1
+
+        su :: forall b. SEXP s b
+        su = R.unsafeCoerce s
 
     case R.typeOf s of
       R.Nil       -> coerce $ return Nil
       R.Symbol    -> coerce $
-        Symbol    <$> (coerceAny <$> R.sexp <$> #{peek SEXPREC, u.symsxp.pname} sptr)
-                  <*> (R.sexp <$> #{peek SEXPREC, u.symsxp.value} sptr)
-                  <*> (R.sexp <$> #{peek SEXPREC, u.symsxp.internal} sptr)
```

Comment:
It looks like `HExp.hsc` doesn't use `hsc2hs` features and can also be promoted to a regular Haskell module now.

Reply:
Renamed.